ios/FluentUI/Core/Theme/FluentTheme.swift

```diff
@@ -47,7 +47,7 @@ import SwiftUI
             return tokens
         } else {
             // Register the default tokens to make future lookups more efficient.
-            let tokens = control.state.defaultTokens
+            let tokens = control.defaultTokens
```

Comment:
Have you tested the scenario where the tokens are overridden for a specific control type (line 47 above) and then had the style/size changed?

Also related to that: why are we registering a new token instance for every combination of style and size?
It seems like the overriding instance should take care of that.

Reply:
Excellent catch. With this, control-type overrides would only apply for a given set of `tokenKeyComponents`.

A solution will tie into your second point: there's no reason that we can't just have a single registered token set per control, and inject whatever overrides are necessary to derive the right token values. In fact this code used to work like that once, but it's been a while since we drifted this (arguably more complex) direction. Let me see if I can steer us back.

Reply:
I believe that if we have the tokens immutable, it probably does not make sense to have a property store their style/size enumerations since can be used only when initializing the properties. Right now they seem to be used only to generate the tokenKeyComponents.
If we have a single overriding type, we can just create it on demand and initialize the tokens based on the style/size params.

Reply:
To be fair, that is effectively what we are doing already. The various token properties are initialized once, the first time they are accessed, as opposed to being generated. The current structure of the token structs is meant for readability: each property's initialization is listed alongside its declaration, rather than all existing as part of a monolithic `initializeVariables()` function. But you're right that such a method would allow for us to remove the style/size storage. Do you have a personal preference?